.ci/lumen_cli/cli/lib/common/utils.py

```diff
@@ -3,7 +3,130 @@
 """
 
 import logging
+import os
+import shlex
+import subprocess
+import sys
+from typing import Optional
 
 
 logger = logging.getLogger(__name__)
-# TODO(elainewy): Add common utils
+
+
+def run_shell(
+    cmd: str,
+    log_cmd: bool = True,
+    cwd: Optional[str] = None,
+    env: Optional[dict] = None,
+    check: bool = True,
+) -> int:
+    """
+    Run a shell command via /bin/bash.
+    Returns the exit code. If check=True and the exit code is non-zero,
+    raises subprocess.CalledProcessError (matching subprocess.run behavior).
+    """
+    if log_cmd:
+        logger.info("[shell] %s", cmd)
+
+    run_env = {**os.environ, **env} if env else None
+
+    proc = subprocess.run(
+        cmd,
+        shell=True,
+        executable="/bin/bash",
+        stdout=sys.stdout,
+        stderr=sys.stderr,
+        cwd=cwd,
+        env=run_env,
+        check=False,  # handle 'check' manually
+    )
+
+    if check and proc.returncode != 0:
+        logger.error("[shell] Command failed (exit %s): %s", proc.returncode, cmd)
+        raise subprocess.CalledProcessError(proc.returncode, cmd)
+
+    return proc.returncode
+
+
+def run_cmd(
+    cmd: str,
+    log_cmd: bool = True,
+    cwd: Optional[str] = None,
+    env: Optional[dict] = None,
+    check: bool = True,
+) -> int:
+    """
+    Run a command using subprocess.run with shell=False.
+
+    Args:
+        cmd[str]: Command string (split with shlex).
+        log_cmd [bool]: Log command before execution.
+        cwd [Optional[str]]: Working directory.
+        env [Optional[dict]]: Environment vars to overlay on current env.
+        check [bool]: If True, raise on non-zero exit; else return code.
+
+    Returns:
+        int: The process's exit code.
+    """
+    args = shlex.split(cmd)
+    if log_cmd:
+        logger.info("[cmd] %s", " ".join(args))
+
+    run_env = {**os.environ, **env} if env else None
```

Comment:
I think the `else` branch means `os.environ` instead of `None`?

```suggestion
    run_env = {**os.environ, **env} if env else os.environ
```